src/services/state-persistor/state-persistor.service.ts

```diff
@@ -0,0 +1,95 @@
+import { watch, watchEffect, WatchStopHandle } from 'vue'
+import { storeToRefs } from 'pinia'
+
+import { useMapStore } from '@/stores/map.store'
+import { Layer } from '@/stores/map.store.model'
+import { useThemeStore } from '@/stores/config.store'
+import useBackgroundLayer from '@/composables/background-layer/background-layer.composable'
+
+import { SP_KEY_BGLAYER, SP_KEY_LAYERS } from './state-persistor.model'
+import { layerMapperService } from './state-persistor-layer.mapper'
+import { storageHelper } from './storage/storage.helper'
+
+class StatePersistorService {
+  bootstrapLayers() {
+    const themeStore = useThemeStore()
+    let stop: WatchStopHandle
+    // eslint-disable-next-line prefer-const
```

Comment:
why not a const btw ? :)

Reply:
To prevent this error at hot reload: 
`runtime-core.esm-bundler.js:557 ReferenceError: Cannot access 'stop' before initialization`